x/evm/types/compiled_contract.go

```diff
@@ -69,51 +72,117 @@ func (s *CompiledContract) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+// HardhatCompiledContract is a type used to unpack the compiled JSON data
+// which is generated by running `npx hardhat compile` for a given smart contract.
+type HardhatCompiledContract struct {
+	Format       string  `json:"_format"`
+	ContractName string  `json:"contractName"`
+	SourceName   string  `json:"sourceName"`
+	ABI          abi.ABI `json:"abi"`
+	Bytecode     string  `json:"bytecode"`
+}
+
+func (c HardhatCompiledContract) ToCompiledContract() (CompiledContract, error) {
+	strippedHex := strings.TrimPrefix(c.Bytecode, "0x")
+	hexBytes, err := hex.DecodeString(strippedHex)
+	if err != nil {
+		return CompiledContract{}, fmt.Errorf("failed to decode hex string: %w", err)
+	}
+
+	return CompiledContract{
+		ABI: c.ABI,
+		Bin: hexBytes,
+	}, nil
+}
+
 var (
-	//go:embed ERC20Contract.json
+	//go:embed testdata/ERC20Contract.json
 	erc20JSON []byte
 
-	// ERC20Contract is the compiled test erc20 contract
-	ERC20Contract CompiledContract
+	//go:embed testdata/SimpleContractHardhat.json
+	SimpleContractJSON []byte
+
+	// SimpleContract is the compiled test simple contract
+	SimpleContract HardhatCompiledContract
 
-	//go:embed SimpleStorageContract.json
+	//go:embed testdata/SimpleStorageContract.json
 	simpleStorageJSON []byte
 
 	// SimpleStorageContract is the compiled test simple storage contract
 	SimpleStorageContract CompiledContract
 
-	//go:embed TestMessageCall.json
+	//go:embed testdata/TestMessageCall.json
 	testMessageCallJSON []byte
 
 	// TestMessageCall is the compiled message call benchmark contract
 	TestMessageCall CompiledContract
 )
 
-func init() {
+func LoadERC20Contract() (CompiledContract, error) {
```

Comment:
I see some of these Load functions are used only on test files
Can we move these to a package related to tests instead of having them on the evm/types pkg?

Same with the embedded contracts